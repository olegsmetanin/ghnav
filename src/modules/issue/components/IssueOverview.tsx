import * as React from 'react'

import { Theme, WithStyles, createStyles, withStyles } from '@material-ui/core/styles'

import Grid from '@material-ui/core/Grid'
import { Loader } from 'common/components/Loader'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import marked from 'marked'

const styles = (theme: Theme) =>
  createStyles({
    root: {
      backgroundColor: theme.palette.background.default,
      marginTop: 30,
      margin: 'auto',
      maxWidth: 600,
      padding: theme.spacing.unit * 2
    },
    paper: {
      marginTop: 30,
      padding: theme.spacing.unit * 2
    },
    content: {
      width: '100%',
      '& pre': {
        overflow: 'auto'
      }
    }
  })

export interface IBaseIssueOverviewProps {
  value: any
  query: any
  owner: string
  repo: string
  num: string
}

export class BaseIssueOverview extends React.Component<IBaseIssueOverviewProps & WithStyles<typeof styles>, {}> {
  render() {
    const { classes, value, query, owner, repo, num } = this.props
    // console.log('render', query, owner, repo, number)
    const isDirty = !(value && query && query.owner === owner && query.repo === repo && query.num === num)

    return (
      <div className={classes.root}>
        {!isDirty && (
          <Paper className={classes.paper}>
            <Grid container spacing={16}>
              <Grid item key="title">
                <Typography variant="h4" color="inherit">
                  {value.title}
                </Typography>
              </Grid>
              <Grid item key="created_at">
                <Typography variant="h6" color="inherit">
                  {value.created_at}
                </Typography>
              </Grid>
              <Grid item key="body" className={classes.content}>
                <div
                  dangerouslySetInnerHTML={{
                    __html: marked(value.body)
                  }}
                />
              </Grid>
            </Grid>
          </Paper>
        )}
        {isDirty && <Loader />}
      </div>
    )
  }
}

export const IssueOverview = withStyles(styles)(BaseIssueOverview)
