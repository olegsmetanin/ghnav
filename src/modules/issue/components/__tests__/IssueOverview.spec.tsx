/* eslint-env jest */
import { IssueOverview } from '../IssueOverview'
import React from 'react'
import renderer from 'react-test-renderer'

describe('IssueOverview', () => {
  it('renders correct', async () => {
    const props = {
      owner: 'owner',
      repo: 'repo',
      num: 'num',
      value: {
        // markdown
        body: 'I have iPhone 5s, 6, 7, 8, XS',
        title: 'title',
        create_at: 'create_at'
      },
      query: {
        owner: 'owner',
        repo: 'repo',
        num: 'num'
      }
    }
    const component = renderer.create(<IssueOverview {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
