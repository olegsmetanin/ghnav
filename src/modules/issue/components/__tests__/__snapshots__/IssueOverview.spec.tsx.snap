// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueOverview renders correct 1`] = `
<div
  className="BaseIssueOverview-root-1"
>
  <div
    className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 BaseIssueOverview-paper-2"
  >
    <div
      className="MuiGrid-container-31 MuiGrid-spacing-xs-16-54"
    >
      <div
        className="MuiGrid-item-32"
      >
        <h4
          className="MuiTypography-root-128 MuiTypography-h4-143 MuiTypography-colorInherit-157"
        >
          title
        </h4>
      </div>
      <div
        className="MuiGrid-item-32"
      >
        <h6
          className="MuiTypography-root-128 MuiTypography-h6-145 MuiTypography-colorInherit-157"
        />
      </div>
      <div
        className="MuiGrid-item-32 BaseIssueOverview-content-3"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>I have iPhone 5s, 6, 7, 8, XS</p>
",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
